packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -489,6 +489,8 @@ class HeroController extends NavigatorObserver {
   }
 
   void _updateQuest(Duration timeStamp) {
+    if (navigator == null)
+      return;
```

Comment:
add a comment explaining that this can be true if the navigator is disposed between the time we received didPush/didPop, and the end of the next frame, since _updateQuest is scheduled to run on a post-frame callback.